File: fix_deepseek_config.py
#!/usr/bin/env python3
"""
修正Deepseek主配置文件 - 确保100%使用Deepseek直连API
修复主目录配置中缺少API Key的问题
"""

import os
import json
import shutil
import sys
from pathlib import Path

def main():
    print("🔧 修正Deepseek主配置文件")
    print("="*50)
    
    # 配置文件路径
    main_config_path = "/Users/zhaoyuzhao/.openclaw/openclaw.json"
    workspace_config_path = "/Users/zhaoyuzhao/.openclaw/xiaofeng_workspace/openclaw.json"
    
    # Deepseek API Key (从工作区配置读取或硬编码)
    DEEPSEEK_API_KEY = "REDACTED_API_KEY"
    
    # 1. 备份原始配置
    backup_path = f"{main_config_path}.backup_before_fix"
    if os.path.exists(main_config_path):
        shutil.copy2(main_config_path, backup_path)
        print(f"📁 备份原始配置: {backup_path}")
    
    # 2. 读取主配置文件
    with open(main_config_path, 'r') as f:
        config = json.load(f)
    
    print(f"📄 当前配置版本: {config.get('meta', {}).get('lastTouchedVersion', 'unknown')}")
    
    # 3. 检查并修正Deepseek profile
    if "auth" not in config:
        config["auth"] = {}
    if "profiles" not in config["auth"]:
        config["auth"]["profiles"] = {}
    
    deepseek_profile = config["auth"]["profiles"].get("deepseek_bryson", {})
    
    print("🔍 当前Deepseek profile状态:")
    print(f"   Provider: {deepseek_profile.get('provider', 'MISSING')}")
    print(f"   Mode: {deepseek_profile.get('mode', 'MISSING')}")
    print(f"   API Key字段: {'✓' if 'apiKey' in deepseek_profile else '✗'} 存在")
    
    # 修正profile
    config["auth"]["profiles"]["deepseek_bryson"] = {
        "provider": "deepseek",
        "type": "apiKey",
        "apiKey": DEEPSEEK_API_KEY
    }
    
    print("✅ Deepseek profile已更新:")
    print(f"   Provider: deepseek")
    print(f"   Type: apiKey")
    print(f"   API Key: {DEEPSEEK_API_KEY[:10]}...")
    
    # 4. 检查xiaofeng agent配置
    xiaofeng_found = False
    for agent in config.get("agents", {}).get("list", []):
        if agent.get("id") == "xiaofeng":
            xiaofeng_found = True
            print(f"🔍 Xiaofeng Agent配置:")
            print(f"   模型: {agent.get('model', 'MISSING')}")
            print(f"   身份: {agent.get('identity', {}).get('name', 'MISSING')}")
            
            # 确保模型是deepseek/deepseek-chat
            if agent.get("model") != "deepseek/deepseek-chat":
                agent["model"] = "deepseek/deepseek-chat"
                print(f"⚠️  修正模型为: deepseek/deepseek-chat")
            break
    
    if not xiaofeng_found:
        print("❌ 未找到xiaofeng agent配置")
    
    # 5. 写入修正后的配置
    with open(main_config_path, 'w') as f:
        json.dump(config, f, indent=2, ensure_ascii=False)
    
    print(f"💾 配置已保存到: {main_config_path}")
    
    # 6. 验证修正
    print("\n🧪 配置文件验证:")
    
    # 读取验证
    with open(main_config_path, 'r') as f:
        new_config = json.load(f)
    
    verified_profile = new_config["auth"]["profiles"].get("deepseek_bryson", {})
    api_key_exists = "apiKey" in verified_profile and len(verified_profile["apiKey"]) > 10
    
    print(f"   API Key存在: {'✅' if api_key_exists else '❌'}")
    print(f"   Provider正确: {'✅' if verified_profile.get('provider') == 'deepseek' else '❌'}")
    
    # 检查模型配置
    xiaofeng_model = None
    for agent in new_config.get("agents", {}).get("list", []):
        if agent.get("id") == "xiaofeng":
            xiaofeng_model = agent.get("model")
            break
    
    print(f"   Xiaofeng模型: {xiaofeng_model or '未找到'}")
    model_correct = xiaofeng_model == "deepseek/deepseek-chat"
    print(f"   模型正确: {'✅' if model_correct else '❌'}")
    
    # 7. 创建回退脚本
    create_restore_script(backup_path, main_config_path)
    
    # 8. 输出下一步建议
    print("\n" + "="*50)
    print("🚀 下一步行动建议:")
    print("1. 重启OpenClaw Gateway使新配置生效")
    print("2. 使用临时测试会话验证Deepseek调用")
    print("3. 检查session_status确认Provider和Key状态")
    print("4. 开始成本跟踪系统优化开发")
    print("\n📋 回退命令:")
    print(f"   cp {backup_path} {main_config_path}")
    print("   或运行: python3 restore_deepseek_fix.py")
    
    return api_key_exists and model_correct

def create_restore_script(backup_path, main_config_path):
    """创建回退脚本"""
    restore_script = """
#!/usr/bin/env python3
# Deepseek配置修复回退脚本

import os
import shutil
import sys

backup_path = "{backup_path}"
main_config_path = "{main_config_path}"

if os.path.exists(backup_path):
    shutil.copy2(backup_path, main_config_path)
    print(f"✅ 配置已回退: {main_config_path}")
    print(f"   从备份恢复: {backup_path}")
else:
    print(f"❌ 备份文件不存在: {backup_path}")
    sys.exit(1)
""".format(backup_path=backup_path, main_config_path=main_config_path)
    
    script_path = "/Users/zhaoyuzhao/.openclaw/xiaofeng_workspace/restore_deepseek_fix.py"
    with open(script_path, 'w') as f:
        f.write(restore_script)
    
    os.chmod(script_path, 0o755)
    print(f"🔧 回退脚本已创建: {script_path}")

if __name__ == "__main__":
    success = main()
    sys.exit(0 if success else 1)

File: fix_deepseek_config.js
#!/usr/bin/env node
const fs = require('fs');
const path = require('path');

const configPath = path.join(process.env.HOME, '.openclaw/openclaw.json');

function fixDeepseekConfig() {
    try {
        const config = JSON.parse(fs.readFileSync(configPath, 'utf8'));
        
        console.log('📋 当前配置:');
        console.log('- xiaofeng agent模型:', config.agents.list.find(a => a.id === 'xiaofeng')?.model);
        console.log('- 默认primary模型:', config.agents.defaults.model.primary);
        console.log('- 默认fallbacks:', config.agents.defaults.model.fallbacks?.join(', '));
        
        // 修复配置
        config.agents.defaults.model.primary = 'deepseek/deepseek-r1';
        config.agents.defaults.model.fallbacks = ['openrouter/deepseek/deepseek-r1'];
        
        // 确保模型配置中包含deepseek直连
        if (!config.agents.defaults.models['deepseek/deepseek-r1']) {
            config.agents.defaults.models['deepseek/deepseek-r1'] = {};
        }
        
        // 保存配置
        fs.writeFileSync(configPath, JSON.stringify(config, null, 4));
        
        console.log('\n✅ 配置已修复:');
        console.log('- 默认primary模型:', config.agents.defaults.model.primary);
        console.log('- 默认fallbacks:', config.agents.defaults.model.fallbacks.join(', '));
        
        // 验证配置
        const envKey = process.env.DEEPSEEK_API_KEY;
        console.log('\n🔑 Deepseek API环境变量:', envKey ? '已设置 (长度:' + envKey.length + ')' : '未设置');
        
        if (!envKey) {
            console.log('⚠️  警告: DEEPSEEK_API_KEY环境变量未设置');
            console.log('   请运行: export DEEPSEEK_API_KEY=你的密钥');
        }
        
        return true;
        
    } catch (error) {
        console.error('❌ 修复配置失败:', error.message);
        return false;
    }
}

function verifyApiKeys() {
    console.log('\n🔍 API密钥验证:');
    console.log('1. Deepseek API Key:', process.env.DEEPSEEK_API_KEY ? '已设置' : '未设置');
    console.log('2. OpenRouter API Key:', process.env.OPENROUTER_API_KEY ? '已设置' : '未设置');
    
    // 检查配置文件中的auth配置
    try {
        const config = JSON.parse(fs.readFileSync(configPath, 'utf8'));
        const authProfiles = config.auth?.profiles || {};
        
        console.log('\n🎯 配置文件验证:');
        for (const [name, profile] of Object.entries(authProfiles)) {
            console.log(`- ${name}: ${profile.provider}/${profile.mode}`);
        }
    } catch (e) {
        console.log('❌ 读取配置文件失败:', e.message);
    }
}

function checkDeepseekModelSupport() {
    console.log('\n🤖 Deepseek直连模型验证:');
    console.log('1. 模型标识: deepseek/deepseek-r1');
    console.log('2. 格式说明: provider/modelName');
    console.log('3. 预期行为: 应该通过Deepseek API密钥直接调用');
    console.log('\n⚠️ 常见问题:');
    console.log('   - 如果出现"Unknown model"，可能是:');
    console.log('     * Deepseek不支持此模型名称');
    console.log('     * API密钥无效');
    console.log('     * 模型名称格式错误');
    console.log('     * OpenClaw版本不兼容');
}

if (require.main === module) {
    console.log('🔧 Deepseek配置修复工具\n');
    checkDeepseekModelSupport();
    verifyApiKeys();
    const success = fixDeepseekConfig();
    
    if (success) {
        console.log('\n🎉 执行下一步:');
        console.log('1. 重启Gateway: export PATH=$PATH:~/.nvm/versions/node/v24.14.0/bin && openclaw gateway restart');
        console.log('2. 验证状态: openclaw status');
        console.log('3. 检查session: session_status');
    } else {
        process.exit(1);
    }
}

module.exports = { fixDeepseekConfig };